Sources/HummingbirdLambda/HBLambdaRequestContext.swift

```diff
@@ -0,0 +1,7 @@
+import Logging
+
+public protocol HBLambdaRequestContext<Event>: HBRequestContext {
+    associatedtype Event
+
+    init(_ event: Event, applicationContext: HBApplicationContext, source: some RequestContextSource, logger: Logger)
+}
```

Comment:
Is it worthwhile adding a `HBBasicLambdaRequestContext` in a similar way to the way we have `HBBasicRequestContext`? Just to get people up and running.

Reply:
I agree, but we can't fill in what Event they're using

Reply:
I'll add a generic `HBBasicLambdaRequestContext<Event>`,

Reply:
Can we not use `HBBasicLambdaRequestContext<Event>`?

Reply:
Hah, yeah that's what I wrote as well. But just wanted to make sure that you were aware we can't completely make it ready off the bat